doc/api/cli.md

```diff
@@ -595,6 +595,31 @@ Path to the file used to store the persistent REPL history. The default path is
 `~/.node_repl_history`, which is overridden by this variable. Setting the value
 to an empty string (`''` or `' '`) disables persistent REPL history.
 
+### `NODE_UNHANDLED_REJECTION=state`
+<!-- YAML
+added: REPLACEME
+-->
+
+Setting this environment variable allows fine grained control over the behavior
+of unhandled rejections. This may be set to either one of `SILENT`, `WARN`,
+`ERROR_ON_GC` or `ERROR` while the default behavior without using the
+environment variable is the same as `WARN`. `WARN` reflects the
+[`unhandledRejection hook`][].
+
+Setting the environment variable to `SILENT` prevents unhandled rejections
+from being logged. The [`unhandledRejection hook`][] is still triggered in case
+it is set.
+
+If set to `ERROR_ON_GC` unhandled rejections that are garbage collected are
+raised as uncaught exceptions.
+
+If set to `ERROR` any unhandled rejection is going to turn the unhandled
```

Comment:
Is there a plan to move towards exiting/aborting synchronously on promises unhandled rejections at some point in the future? If so, would that behavior be introduced with a different name, like `SYNCHRONOUS_ERROR`?

Reply:
@misterdjules the problem is that `Promise.reject('foo').catch(() => ...)` creates an error synchronously and only then handles it. 

We don't have a V8 hook that gets called synchronously as far as I know - `ERROR` happens before any other microtick which loses _a little_ but not much information.

I intend to use `ERROR` myself in my own code to be an even stricter `process.on('unhandledRejection', e => { throw e })` which I've been using.

Hopefully, combining it with tooling like Anna's stack-trace-for-rejection module will allow people to get a _reasonable_ postmortem experience. More ideas/suggestions welcome - the reason Anna stopped working on these is because there wasn't much community interest - so community interest could help here.

Reply:
I was asking this question because I thought I had heard @BridgeAR mention that he was working on an implementation that would exit synchronously on unhandled rejections. Maybe that was a misunderstanding.